tests/integration/targets/setup_os_pkg_name/tasks/alpine.yml

```
@@ -0,0 +1,11 @@
+---
+# Copyright (c) Ansible Project
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+- name: Update OS Package name fact (alpine)
+  ansible.builtin.set_fact:
+    os_package_name: "{{ os_package_name | combine({item.key: item.value}) }}"
+  loop:
+    - key: virtualenv
+      value: py3-virtualenv
```

Why not simply:
```suggestion
  ansible.builtin.set_fact:
    os_package_name: "{{ os_package_name | combine(specific_package_names) }}"
  vars:
    specific_package_names:
      virtualenv: py3-virtualenv
```
?

That's similar readable and more efficient (since you don't need to loop).